pyaedt/application/Analysis3D.py

```diff
@@ -1277,6 +1277,117 @@ def import_dxf(
+            Number of element must be the same as gds_number list.
+            If not or None elevation and thickness will not be applied.
+        import_method : integer, optional
+            GDSII import method. The default is 1
+            0 = script, 1 = Parasolid
+
+                Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+
+        if not os.path.exists(gds_file_path):
+            self.logger.error("GDSII file does not exist, no layer imported ")
+            return False
+        if len(gds_number_list) == 0:
+            self.logger.error("GDSII layer number list is empty, no layer imported ")
+            return False
+
+        if (
+            elevation_list is None
+            or thickness_list is None
+            or (len(gds_number_list) != len(elevation_list) or len(gds_number_list) != len(thickness_list))
+        ):
+            self.logger.warning(
+                "One of parameter list: elevation and/or thickness is either: "
+                "not specified or has different number of element than GDS number."
+                "Elevation and thickness will not be applied "
+            )
+            layermap = ["NAME:LayerMap"]
+            for i in gds_number_list:
+                layername = "signal" + str(i)
+                layermap.append(
+                    ["NAME:LayerMapInfo", "LayerNum:=", i, "DestLayer:=", layername, "layer_type:=", "signal"]
+                )
+
+            self.oeditor.ImportGDSII(
+                [
+                    "NAME:options",
+                    "FileName:=",
+                    gds_file_path,
+                    "FlattenHierarchy:=",
+                    True,
+                    "ImportMethod:=",
+                    import_method,
+                    layermap,
+                ]
+            )
+        else:
+            self.logger.warning("GDS layer imported with elevation and thickness")
+
+            layermap = ["NAME:LayerMap"]
+            ordermap = []
+            j = 0
+            for i in gds_number_list:
+                layername = "signal" + str(i)
+                layermap.append(
+                    ["NAME:LayerMapInfo", "LayerNum:=", i, "DestLayer:=", layername, "layer_type:=", "signal"]
+                )
+                ordermap1 = [
+                    "entry:=",
+                    [
+                        "order:=",
+                        j,
+                        "layer:=",
+                        layername,
+                        "LayerNumber:=",
+                        i,
+                        "Thickness:=",
+                        thickness_list[j],
+                        "Elevation:=",
+                        elevation_list[j],
+                        "Color:=",
+                        "color",
+                    ],
+                ]
+                ordermap = ordermap + ordermap1
+                j = j + 1
```

Comment:
@amichel0205 if you have to work with indeces, why don't you use a for loop as:
for(j=0; j<(something);j++) ?

Reply:
How I can combine:

 for i in gds_number_list: 

and indices?

Reply:
@amichel0205 gds_number is a list so you can access it through indeces: gds_number[j]

Reply:
Giulia, can you please check the syntax is correct?

C:\AnsysDev\NoBackup\venvs\venv_pyaedt_dev\Scripts\python.exe C:\Users\amichel\AppData\Roaming\JetBrains\PyCharmCE2023.3\scratches\tes_nested_list.py 
  File "C:\Users\amichel\AppData\Roaming\JetBrains\PyCharmCE2023.3\scratches\tes_nested_list.py", line 19
    for(i=0;i<(len(gds_number));i++):
        ^^^
SyntaxError: invalid syntax. Maybe you meant '==' or ':=' instead of '='?

Reply:
I found the solution: for:

 i in range(len(gds_number)):

Reply:
done